### ISSUE TITLE
Taxonomies Term Localization

### ISSUE THREAD

**Issue by urbanit** [OrchardCMS/OrchardCore#4845]
Create a ContentType: `CategoryTerm` with Title, Localization Part, AutoRoute
Create a Taxonomy: `Categories` with terms of `CategoryTerm`
Add `CategoryTerm` to taxonomy --> a term is created
Edit term & add localized version --> **Error!**
`No webpage was found for the web address: http://{url}/OrchardCore.ContentLocalization/Admin/Localize`

**Comment by sebastienros** [comment]
We can't localize terms, only taxonomies

**Comment by urbanit** [comment]
How we 'sync' terms? Meaning Term_1_Gr -> Term_1_En and vice versa?

**Comment by sebastienros** [comment]
I talked about it with @jptissot and we had found a way, he will remember I am sure

**Comment by urbanit** [comment]
Cool, I will get an answer from @jptissot and then I will close the issue...

**Comment by dodyg** [comment]
How do you resolve situation where you have a ContentItem that are multi-lingual that has a `Taxonomy Field`? 

![taxonomy](https://user-images.githubusercontent.com/1761249/71308870-2d822c00-240a-11ea-84b1-7b2841f9ad17.png)

**Comment by dodyg** [comment]
@urbanit did you ever found a workaround for this issue?

**Comment by dodyg** [comment]
My workaround is to put a `Language field` in the term and filter the term using `where` filter based on this when showing the taxonomy term list. The downside is that in the admin you have double the length for the terms (English and Arabic).

**Comment by urbanit** [comment]
@dodyg until now, the guides are that we add localization part to taxonomies and you add all localized taxonomies to your content type. Each time you create a content item you check the term from localized tazonomy... What we miss here is that termX(langA) and termX(langB) are kinda of "translated" terms...

**Comment by dodyg** [comment]
The problem with localized taxonomy is that it fails at "Taxonomy Field". Taxonomy field only accepts one version. 

So if you have a taxonomy in three languages, you will need three "Taxonomy Fields", one for each language -- which  is silly.

**Comment by jptissot** [comment]
I encountered the same issue. We either need a LocalizedTaxonomyPicker Field or change the options in the current picker to allow selecting a Localized Taxonomy

**Comment by sebastienros** [comment]
Or can we select the localized version automatically if it exists? From the culture of the currently edited content item?

**Comment by urbanit** [comment]
- [x] Localized Version of Taxonomy with #6316 
- [ ] Add Alias to sync terms of localized taxonomies
- [ ] Extend CulturePicker to support Term Aliases

**Comment by pbros** [comment]
Any updates on this?

**Comment by urbanit** [comment]
Not yet, probably this week (at least for the PR)

**Comment by Lawyerson** [comment]
As I understand from watching the Taxonomy localization demo, there are no plans to provide Term localization out of the box?
Term localization is an important feature to me and seems fairly trivial to implement on top of Taxonomy localization. I think it can be done with minimal code, but correct me if I'm missing something:

1. Attach LocalizationPart to the ContentType that is used as a Term.
2. Write a handler to ensure that localized content items with a TermPart get the TaxonomyContentItemId of the localized version of their master content item's Taxonomy upon creation.
3. Provide an alternate editor for the LocalizationPart on unlinked localized Terms to select a master content item from available Terms under the master Taxonomy.

I think all of this can be done without touching the source but 3. might be a little hairy.

**Comment by sebastienros** [comment]
@Lawyerson you are correct. And we encourage you to work on that if you are willing to.

**Comment by Skrypt** [comment]
Anyone made progress on that one?

**Comment by sebastienros** [comment]
We can't add LocalizationPart on MenuItems or Terms, because they are embedded. One solution might be to have first-class localization support on these types, and they would store/edit the localized values by themselves, handling the UI.

**Comment by urbanit** [comment]
My way on this is to keep the same alias on terms of different culture in order to correlate them. Maybe we can use it in as workaround and/or expose it somehow.

**Comment by dohoangtung** [comment]
> My workaround is to put a `Language field` in the term and filter the term using `where` filter based on this when showing the taxonomy term list. The downside is that in the admin you have double the length for the terms (English and Arabic).

Hi @dodyg , can you share your code snippet for that? Thanks!

**Comment by hishamco** [comment]
I came across this on my current website, you can add a `LocalizationPart` to the `Taxonomy` content type, then you can flip the alias when you render the term shape
